modules/Auth/lib/AuthHandler.cpp

```diff
@@ -161,6 +163,7 @@ TokenHandlingResult AuthHandler::handle_token(const ProvidedIdToken& provided_to
         while (i < validation_results.size() && !authorized && !referenced_connectors.empty()) {
             auto validation_result = validation_results.at(i);
             if (validation_result.authorization_status == AuthorizationStatus::Accepted) {
+                this->publish_token_validation_status_callback(types::authorization::TokenValidationStatus::Accepted);
```

Comment:
maybe we should move this out of the while loop

```cpp
if (!referenced_connectors.empty())
  this->publish_token...
```

Reply:
The blocking part is next that happens. The only way we can possible do it is to get it out of the while loop and write sth like that:

```
if (!referenced_connectors.empty() and std::any_of(validation_results.cbegin(), validation_results.cend(),  [](const& auto validation_result) {
    return validation_result.authorization_status == AuthorizationStatus::Accepted
}) {
    this->publish_token...
}


// while(...)
```

Reply:
Oh yes, correct.. and duplicating this filter would be quite ugly.

The better way would be then to move this filter out of the while loop and use https://en.cppreference.com/w/cpp/algorithm/remove

This would make the code way more readable but lets contain the changes to a min